MmSupervisorPkg/Core/Handler/Mmi.c

```diff
@@ -228,15 +283,78 @@ MmiManage (
         ASSERT_EFI_ERROR (Status);
         break;
     }
+
+    if (WillReturn) {
+      break;
+    }
   }
 
-  if (SuccessReturn) {
-    Status = EFI_SUCCESS;
+  ASSERT (mMmiManageCallingDepth > 0);

```

Comment:
PanicLib will catch this even with asserts disabled. Should this get switched over to use it instead?

Reply:
Updated.